tests/kernel/context/src/context.c

```diff
@@ -934,46 +928,38 @@ void main(void)
 	TC_PRINT("Thread to test k_current_get() and " "k_is_in_isr()\n");
 	k_sem_give(&sem_thread);
 
-	if (thread_detected_error != 0) {
-		rv = TC_FAIL;
-		TC_ERROR("  - failure detected in thread; "
-			 "thread_detected_error = %d\n", thread_detected_error);
-		goto tests_done;
-	}
+	zassert_equal(thread_detected_error, 0,
+		      "  - failure detected in thread; "
+		      "thread_detected_error = %d\n", thread_detected_error);
 
 	TC_PRINT("Thread to test k_yield()\n");
 	k_sem_give(&sem_thread);
 
-	if (thread_detected_error != 0) {
-		rv = TC_FAIL;
-		TC_ERROR("  - failure detected in thread; "
-			 "thread_detected_error = %d\n", thread_detected_error);
-		goto tests_done;
-	}
+	zassert_equal(thread_detected_error, 0,
+		      "  - failure detected in thread; "
+		      "thread_detected_error = %d\n", thread_detected_error);
 
 	k_sem_give(&sem_thread);
 
 	rv = test_timeout();
-	if (rv != TC_PASS) {
-		goto tests_done;
-	}
+	zassert_equal(rv, TC_PASS, "test timeout\n");
 
 #ifdef HAS_POWERSAVE_INSTRUCTION
 	TC_PRINT("Testing k_cpu_idle()\n");
 	rv = test_kernel_cpu_idle(0);
-	if (rv != TC_PASS) {
-		goto tests_done;
-	}
+	zassert_equal(rv, TC_PASS, "failure test kernel cpu idel");
 #ifndef CONFIG_ARM
 	TC_PRINT("Testing k_cpu_atomic_idle()\n");
 	rv = test_kernel_cpu_idle(1);
-	if (rv != TC_PASS) {
-		goto tests_done;
-	}
+	zassert_equal(rv, TC_PASS, " failure kernel cpu idle");
 #endif
 #endif
+	k_thread_priority_set(k_current_get(), old_prio);
```

Comment:
why is this needed?
ztest creates a new thread for every case so I don't see the point

look at tests/ztest/src/ztest.c run_test()

Reply:
yes I know that ztest creates a thread with priority -1. I tried to preserve priority after setting it to 0 as it was done in other test cases. But I don't mind removing it.